### PR TITLE
update to rc.7 and use vitejs/plugin-react-swc

### PR DESCRIPTION
**Pull request by jwartofsky-yext**
Updates all dependencies except react dependencies to latest version.

Tested npm run dev and npm run prod. Verified that turtlehead-tacos pages appear as expected.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,44 +6,45 @@
   "license": "BSD-3-Clause",
   "type": "module",
   "scripts": {
-    "dev": "yext pages build && yext pages generate-test-data && pages dev",
-    "build:local": "pages build",
-    "build": "yext pages build && yext pages generate-test-data && yext pages render",
+    "dev": "pages dev",
+    "build": "pages build",
     "build:serve": "npm run build && yext pages serve",
```
Comment (asanehisa) [inline package.json]: simplify the scripts 

  "scripts": {
    "dev": "pages dev",
    "prod": "pages prod",
    "build": "pages build"
  },

Comment (jwartofsky-yext) [reply package.json]: Done

File: package.json [hunk 2]
```diff
@@ -6,44 +6,38 @@
...
   "devDependencies": {
     "@types/react": "^17.0.49",
     "@types/react-dom": "^17.0.17",
-    "@typescript-eslint/eslint-plugin": "^5.36.2",
-    "@typescript-eslint/parser": "^5.36.2",
-    "@vitejs/plugin-react": "^2.1.0",
-    "@yext/pages": "1.0.0-beta.21",
+    "@typescript-eslint/eslint-plugin": "^6.16.0",
+    "@typescript-eslint/parser": "^6.16.0",
+    "@vitejs/plugin-react": "4.2.1",
```
Comment (asanehisa) [inline package.json]: ^

File: package.json [hunk 3]
```diff
@@ -6,44 +6,38 @@
...
+    "@yext/pages": "1.0.0-rc.7",
```
Comment (asanehisa) [inline package.json]: ^

File: package.json [hunk 4]
```diff
@@ -6,44 +6,38 @@
...
+    "@yext/pages-components": "1.0.0-rc.0",
```
Comment (asanehisa) [inline package.json]: ^

**Review by asanehisa — COMMENTED** [review]
Also remove generate-notices from github actions

**Review by asanehisa — APPROVED** [review]

File: .gitignore [hunk 5]
```diff
@@ -4,6 +4,8 @@ localData/
 dist/
 .yext
 .artifact-output
+.env
+!.env.local
```
Comment (mkilpatrick) [inline .gitignore:8]: Is this line necessary?

Comment (jwartofsky-yext) [reply .gitignore:8]: Yes, if we don't have it then .env.local gets ignored

Comment (asanehisa) [reply .gitignore:8]: Hmmm I think we wanna ignore .env.local normally?

Comment (mkilpatrick) [reply .gitignore:8]: We want .env.local to be checked in

Comment (jwartofsky-yext) [reply .gitignore:8]: .env with !.env.local means that .env gets ignored, but .env.local does not get ignored which is what we want

**Review by mkilpatrick — APPROVED** [review]

**Pull request merged**